### PR TITLE
Alert user when trying to run in a DomU (bsc#952253)

### PR DESCRIPTION
**Pull request by ancorgs**

### CODE REVIEW

File: src/include/kdump/uifunctions.rb [hunk 1]
```diff
@@ -1844,5 +1844,15 @@ def ValidEmail(key, event)
       Popup.Message(key)
       true
     end
+
+    def unsupported_kdump_confirmation
+      text = _(
+        "Kdump is not supported on this system.\n" \
+        "If you continue now, the module may not function properly.\n" \
+        "For example, some settings will not be correctly read\n" \
+        "and the resulting configuration will probably be useless.\n"
```
Comment (jreidinger) [inline src/include/kdump/uifunctions.rb]: maybe "will be probably useless"? But your english is usually better ;)

File: src/lib/kdump/kdump_calibrator.rb [hunk 2]
```diff
@@ -100,14 +102,10 @@ def max_high
...
-      return @total_memory if @total_memory
-
-      # Calculating the total memory in a system with kdump enabled is tricky.
       # As a best effort if kdumptool is not available, let's use the physical
       # memory reported by the kernel.
-      probe = SCR.Read(Yast::Path.new(".probe.memory"))
-      resource = probe.first["resource"]
-      @total_memory = resource["phys_mem"][0]["range"] / MiB_SIZE
+      @total_memory = system.reported_memory if @total_memory.nil?
+      @total_memory
```
Comment (jreidinger) [inline src/lib/kdump/kdump_calibrator.rb]: why not simple `@total_memory ||= system.reported_memory`

File: src/lib/kdump/kdump_system.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
+require "yast"
+
+module Yast
+  Yast.import "Arch"
+
+  class KdumpSystem
+    MiB_SIZE = 1048576
```
Comment (jreidinger) [inline src/lib/kdump/kdump_system.rb]: btw I think rubocop suggest to use `1_048_576` for easier number reading.

File: src/lib/kdump/kdump_system.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    HYPER_TYPE_FILE = "/sys/hypervisor/type"
+    HYPER_FEATURES_FILE = "/sys/hypervisor/properties/features"
+    XENFEAT_DOM0 = 11
```
Comment (jreidinger) [inline src/lib/kdump/kdump_system.rb]: what is XENFEAT? I think better name helps

File: src/lib/kdump/kdump_system.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    #
+    # Calculating the total memory in a system with kdump enabled is tricky,
+    # since the amount reported by the kernel by the normal methods will be
+    # lower than the real total memory. Thus, use this method only if you don't
+    # have a more precise method (like calling kdumptool) available
+    def reported_memory
+      return @reported_memory if @reported_memory
+
+      probe = SCR.Read(Path.new(".probe.memory"))
+      resource = probe.first["resource"]
```
Comment (jreidinger) [inline src/lib/kdump/kdump_system.rb]: are you sure SCR cannot return nil?

File: src/lib/kdump/kdump_system.rb (hunk 4 above)
Comment (jreidinger) [inline src/lib/kdump/kdump_system.rb]: after reading code and getting how it work I propose to call it `DOM0_FLAG`

File: src/lib/kdump/kdump_system.rb [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    end
+
+    # Checks whether the system is a DomU
+    #
+    # See bsc#952253 for the rationale of the implementation
+    def domU?
+      domU = false
+      type = SCR.Read(Path.new(".target.string"), HYPER_TYPE_FILE)
+      # Check if the system is a Xen domain
+      if type && type.strip.downcase == "xen"
```
Comment (jreidinger) [inline src/lib/kdump/kdump_system.rb]: better use `Arch.is_xen`

Comment (jreidinger) [inline src/lib/kdump/kdump_system.rb]: and whole method can be replaced by `Arch.is_xenU` see  https://github.com/yast/yast-yast2/blob/fcc522455ed8ce4de011642be05b9f2084980bcd/library/general/src/modules/Arch.rb#L429

File: src/modules/Kdump.rb [hunk 7]
```diff
@@ -966,7 +972,7 @@ def Import(settings)
     #
     # @return [Boolean] is supported
     def fadump_supported?
```
Comment (jreidinger) [inline src/modules/Kdump.rb]: does it make sense to have that method at all?

File: src/include/kdump/uifunctions.rb (hunk 1 above)
Comment (ancorgs) [inline src/include/kdump/uifunctions.rb]: Both are ok, I'd say.

File: src/lib/kdump/kdump_system.rb (hunk 5 above)
Comment (ancorgs) [inline src/lib/kdump/kdump_system.rb]: Pretty much, but I can add an extra check returning 0.

File: src/modules/Kdump.rb (hunk 7 above)
Comment (ancorgs) [inline src/modules/Kdump.rb]: Well, it's used from the clients and from the UI and I though that maybe it was worth keeping it as a way to "hide" implementation details to the layers on top of the Kdump YCP-style module. But I guess I could also simply substitute all the calls with `Kdump.system.supports_fadump?`

Comment (ancorgs) [inline src/modules/Kdump.rb]: Since I got 2 comments about that and the method is actually not published, I will do the change.

**Comment by jreidinger** [comment]
LGTM

**Comment by cwh42** [comment]
Also LGTM

**Pull request merged**